### PR TITLE
Update Experience Event class for Datastream Config Override Feature

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edge/src/main/java/com/adobe/marketing/mobile/ExperienceEvent.java [hunk 1]
```diff
@@ -42,6 +53,33 @@ public Builder() {
...
+		 * @throws UnsupportedOperationException if this instance was already built
+		 */
+		public Builder setDatastreamIdOverride(final String datastreamIdOverride) {
+			throwIfAlreadyBuilt();
+			experienceEvent.datastreamIdOverride = datastreamIdOverride;
+			return this;
+		}
+
+		/**
+		 * Sets configuration to override default Datastream configurations.
```
Comment (kevinlind) [inline code/edge/src/main/java/com/adobe/marketing/mobile/ExperienceEvent.java:70]: Can you expand on this description? Something like:

> Override the default datastream configuration to trigger different datastream behaviors than the default ones for this event.
> 
> When using {@link Edge#sendEvent}, this event is sent to the Experience Platform along with the datastream overrides defined in {@code datastreamConfigOverride}.
>

File: code/edge/src/main/java/com/adobe/marketing/mobile/ExperienceEvent.java [hunk 2]
```diff
@@ -42,6 +53,33 @@ public Builder() {
 			didBuild = false;
 		}
 
+		/**
+		 * Sets the Datastream identifier to override the default Datastream identifier set in the Edge configuration.
+		 *
+		 * @param datastreamIdOverride Datastream identifier to override the default Datastream identifier set in the Edge configuration
```
Comment (kevinlind) [inline code/edge/src/main/java/com/adobe/marketing/mobile/ExperienceEvent.java:59]: Datastream shouldn't be capitalized (unless it's starting a sentence 😄 )

File: code/edge/src/main/java/com/adobe/marketing/mobile/ExperienceEvent.java [hunk 3]
```diff
@@ -42,6 +53,33 @@ public Builder() {
...
+		public Builder setDatastreamIdOverride(final String datastreamIdOverride) {
+			throwIfAlreadyBuilt();
+			experienceEvent.datastreamIdOverride = datastreamIdOverride;
+			return this;
+		}
+
+		/**
+		 * Sets configuration to override default Datastream configurations.
+		 *
+		 * @param datastreamConfigOverride configuration map to override default Datastream configurations with
```
Comment (kevinlind) [inline code/edge/src/main/java/com/adobe/marketing/mobile/ExperienceEvent.java:72]: Suggest rewording: "Map defining datastream configuration overrides for this Experience Event"

File: code/edge/src/test/java/com/adobe/marketing/mobile/ExperienceEventTests.java
Comment (kevinlind) [inline code/edge/src/test/java/com/adobe/marketing/mobile/ExperienceEventTests.java:1]: Please add some tests to cover ExperienceEvent.getDatastreamIdOverride() and ExperienceEvent.getDatastreamConfigOverride(). For getDatastreamConfigOverride, look at the tests which verify changing the Map after setting to the ExperienceEvent won't change the values in the ExperienceEvent.

**Review by kevinlind — COMMENTED** [review]
Are you planning on deprecating the existing APIs which allow setting the dataset ID outside of the datastream configuration override?

**Comment by addb** [comment]
> Are you planning on deprecating the existing APIs which allow setting the dataset ID outside of the datastream configuration override?

Yes, will deprecate the setDataset in a separate task, after completing this config override feature completion.

**Review by kevinlind — APPROVED** [review]
